docs/source/onnx_dynamo.md

```diff
@@ -143,6 +143,124 @@ to resolve the issue.
 :hidden:
 onnx_dynamo_memory_usage
 ```
+## Metadata
+
+During ONNX export, each ONNX node is annotated with metadata that helps trace its origin and context from the original PyTorch model. This metadata is useful for debugging, model inspection, and understanding the mapping between PyTorch and ONNX graphs.
+
+The following metadata fields are added to each ONNX node:
+
+- **namespace**
+
+  A string representing the hierarchical namespace of the node, consisting of a stack trace of modules/methods.
+
+  *Example:*
+  `_empty_nn_module_stack_from_metadata_hook: _empty_nn_module_stack_from_metadata_hook/sym_size_int_720: aten.sym_size.int`
+
+- **pkg.torch.onnx.class_hierarchy**
+
+  A list of class names representing the hierarchy of modules leading to this node.
+  
+  *Example:*
+  `['_empty_nn_module_stack_from_metadata_hook', 'aten.sym_size.int']`
+
+- **pkg.torch.onnx.fx_node**
+
+  The string representation of the original FX node, including its name, number of consumers, the targeted torch op, arguments, and keyword arguments.
+  
+  *Example:*
+  `%sym_size_int_720 : [num_users=1] = call_function[target=torch.ops.aten.sym_size.int](args = (%img, 2), kwargs = {})`
+
+- **pkg.torch.onnx.name_scopes**
+
+  A list of name scopes (methods) representing the path to this node in the PyTorch model.
+
+  *Example:*
+  `['_empty_nn_module_stack_from_metadata_hook', 'sym_size_int_720']`
+
+- **pkg.torch.onnx.stack_trace**
+
+  The stack trace from the original code where this node was created, if available.
+
+  *Example:*
+  `File "torch/fx/passes/runtime_assert.py", line 24, in insert_deferred_runtime_asserts`
+
+- **"pkg.torch.onnx.input_names**
+
+  The inputs for a node:
+
+  *Example:*
+  `[2]`
```

Comment:
```suggestion
  `[2]`
```

can you share where it shows up? Strange to have a number as input

Reply:
Should be here: https://github.com/pytorch/pytorch/blob/f402eed4d9515dee5bb9b11a2b157de57a5988c1/torch/onnx/_internal/exporter/_core.py#L419

Reply:
You are right this is a path we don't expose to users. Feel free to remove